dask/bag/core.py

```diff
@@ -1082,6 +1082,51 @@ def from_sequence(seq, partition_size=None, npartitions=None):
     return Bag(d, name, len(d))
 
 
+def from_castra(x, columns=None, index=False, npartitions=None):
+    """Load a dask Bag from a Castra.
+
+    Parameters
+    ----------
+    x : filename or Castra
+    columns: list or string, optional
+        The columns to load. Default is all columns.
+    index: bool, optional
+        If True, the index is included as the first element in each tuple.
+        Default is False.
+    npartitions: int, optional
+        The number of desired partitions. Defaults to number of partitions in
+        the Castra.
+    """
+    from castra import Castra
+    if not isinstance(x, Castra):
+        x = Castra(x)
+    path = x.path
+    if columns is None:
+        columns = x.columns
+    if npartitions is None:
+        npartitions = len(x.partitions)
+    parts = from_sequence(x.partitions, npartitions=npartitions)
+    func = lambda p: load_castra_partition(path, p, columns, index)
+    return parts.map_partitions(func).map_partitions(list)
+
+
+def load_castra_partition(path, parts, columns, index):
+    from castra import Castra
+    import blosc
+    # Due to serialization issues, blosc needs to be manually initialized in
+    # each process.
+    blosc.init()
```

Comment:
This works and has negligible overhead. However, it feels hacky. Without it though, blosc locks up due to being uninitialized in the subprocesses. There's some funky serialization issue going on here, as castra is being serialized, and using blosc, which isn't locally initialized in each process. It would be nice to have a way to determine if blosc has been initialized in the process, and only run that once for each process. 

@esc, do you have any thoughts on this? Is there any harm in running `blosc.init()` multiple times in the same process?